### PR TITLE
refactor execute mars to update context

### PR DESCRIPTION
**Pull request by malmans2**

### CODE REVIEW

**Comment by EddyCMWF** [comment]
Hi @malmans2 ,
This looks really good, and probably what we should have done first time round!
I'm just going to merge into the develop branch so we can have a quick test on some real datasets from a real website, then I'll review.
Eddy

**Comment by malmans2** [comment]
OK, I'm closing #55

File: cads_adaptors/adaptors/multi.py [hunk 1]
```diff
@@ -120,11 +120,7 @@ def retrieve(self, request: Request):
                 )
 
         logger.debug(f"MultiMarsCdsAdaptor, mapped_requests: {mapped_requests}")
-        result, output = execute_mars(mapped_requests)
-
-        self.context.stdout = output.stdout.decode()
-        self.context.stderr = output.stderr.decode()
-        self.context.user_visible_log = output.stdout.decode()
+        result = execute_mars(mapped_requests, context=self.context)
```
Comment (EddyCMWF) [inline cads_adaptors/adaptors/multi.py:123]:
```suggestion
        result, _ = execute_mars(mapped_requests, context=self.context)
```

Comment (EddyCMWF) [reply cads_adaptors/adaptors/multi.py:123]: Alternatively we revert execute_mars to only return the result as we no longer use the output... outside of execute_mars

**Comment by EddyCMWF** [comment]
Hi @malmans2 ,
I've checked this (with my suggested fix) on ecds-dev with the standard and multi Mars adaptor and it's working fine.
So if you add my suggestion I am happy to accept and merge.
Eddy

**Review by EddyCMWF — CHANGES_REQUESTED** [review]
address the mishandling of the output in the multi mars adaptor

File: cads_adaptors/adaptors/multi.py (hunk 1 above)
Comment (malmans2) [reply cads_adaptors/adaptors/multi.py:123]: OK, I'll return result only. I wasn't sure if that function is used elsewhere and didn't want to introduce breaking changes.

**Review by EddyCMWF — APPROVED** [review]

**Pull request merged**